cmd/promremotecli.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/m3db/m3/src/query/ts"
+	"github.com/m3db/prometheus_remote_client_golang/promremote"
+)
+
+var (
+	writeURL    *string
+	tagsListVar tagList
+	dpVar       dp
+)
+
+type tagList []promremote.Tag
+type dp ts.Datapoint
+
+func init() {
+	writeURL = flag.String("u", promremote.DefaultRemoteWrite, "remote write endpoint")
+	flag.Var(&tagsListVar, "t", "tag pair to include in metric. specify as key:value e.g. status_code:200")
+	flag.Var(&dpVar, "d", "datapoint to add. specify as value(float),unixTimestamp(int) e.g. 14.23,1556026059. use `now` instead of timestamp for current time")
```

Comment:
Better to get rid of the `init` and call `flag.Parse()` inside of the `main` itself.

`init` is really only supposed to be used when you absolutely can't get around not using it.  This is because it will slow down your application startup speed since every single package needs to run it's `init`.  So if a ton of your dependencies has `init` methods defined your binary has a fixed slow cost to it's startup.

Also I would avoid mixing `flag.FooVar` and `flag.Foo` patterns, i.e. choose one or the other, so better here to probably use `flag.StringVar` instead of just `flag.String` to match the others.

You can do what you want to do like this:
```go

var (
  writeURLFlag string
  tagListFlag tagList
  dpFlag dp
)

func main() {
	flag.StringVar(&writeURLFlag, "u", promremote.DefaultRemoteWrite, "remote write endpoint")
	flag.Var(&tagsListFlag, "t", "tag pair to include in metric. specify as key:value e.g. status_code:200")
	flag.Var(&dpFlag, "d", "datapoint to add. specify as value(float),unixTimestamp(int) e.g. 14.23,1556026059. use `now` instead of timestamp for current time")
	flag.Parse()

	// now use the vars
}
```